types.ts

```diff
@@ -253,10 +253,12 @@ export interface BaseAuthentication {
   postSetup?: PostSetup[];
 
   /**
-   * Which domain should get auth credentials, when a pack is configured with multiple domains.
+   * Which domain(s) should get auth credentials, when a pack is configured with multiple domains.
    * Packs configured with only one domain or with requiredsEndpointUrl set to true can omit this.
+   *
+   * Using multiple authenticated network domains is uncommon and requires Coda approval.
    */
-  networkDomain?: string;
+  networkDomain?: string | string[];
```

Comment:
btw. is anyone (external) actually using this? if not we can probably just make a breaking change? @jonathan-codaio

Reply:
I suspect it's only the unreleased Box pack which needs multi-domain auth and would require changes anyway, but potentially anyone with multiple domains who switched to SDK 0.11.0 

Not sure if there's any quick and easy way to search for packs with this in the metadata, but we could scan s3

Reply:
not a big deal though. just calling out in case if it's easy.

Reply:
Oh, actually validation is currently for SDK versions `>=1.0.0`, so probably the Box pack is the only one that even noticed this existed -- before we added the version-based verification. Seems likely that that pack + first party packs are the only ones using this field

Reply:
Personally I'm comfortable making this a breaking change since it affects essentially nobody who doesn't have to update their config anyway

The change would presumably be to rename the field to `networkDomains` with an "s" and make type always an array `string[]`? This would only affect people using multiple network domains, since packs with a single domain can omit the field entirely

@jonathan-codaio and @ekoleda-codaio, thoughts?

Reply:
Probably wouldn't affect anybody, but on principle considering we just announced the new field yesterday, we shouldn't go break it right away.

I actually think the non-breaking `networkDomain?: string | string[];` might be the best interface anyway--for the vast majority of developers they will either omit this or provide a single domain as a string. It's only for really really rare cases that it needs to be an array, so it would kind of be the tail wagging the dog if we designed the interface around that.

What we're learning from our makers is that the conceptually correct interface isn't really as helpful as one that kinds them towards common behaviors.

Reply:
+1 to Jonathan's points.